File: SPCServer/springboot/src/main/java/com/lcaohoanq/shoppe/domain/auth/IAuthService.java
package com.lcaohoanq.shoppe.domain.auth;

import com.lcaohoanq.shoppe.domain.user.UserResponse;
import com.lcaohoanq.shoppe.domain.user.User;
import java.util.List;

public interface IAuthService {

    User register(AccountRegisterDTO accountRegisterDTO) throws Exception;
    String login(String email, String password) throws Exception;
    UserResponse getUserDetailsFromToken(String token) throws Exception;
    void logout(String token, User user) throws Exception;
    void verifyOtpToVerifyUser(Long userId, String otp) throws Exception;
    void verifyOtpIsCorrect(Long userId, String otp) throws Exception;
}

File: spcserver/kotlin/src/main/kotlin/com/lcaohoanq/kotlinbasics/domain/auth/IAuthService.kt
package com.lcaohoanq.kotlinbasics.domain.auth

import com.lcaohoanq.kotlinbasics.domain.token.TokenPort
import com.lcaohoanq.kotlinbasics.domain.user.User
import com.lcaohoanq.kotlinbasics.domain.user.UserPort

interface IAuthService {

    fun login(account: AuthPort.AuthRequest): AuthPort.AuthResponse
    fun register(newAccount: AuthPort.SignUpReq)
    fun getUserDetailsFromToken(token: String): UserPort.UserResponse
    fun getCurrentAuthenticatedUser(): User
    fun refreshToken(refreshTokenDTO: TokenPort.RefreshTokenDTO): AuthPort.AuthResponse
    fun logout(token: String, user: User)
}